packages/web3-providers/src/opensea/index.ts

```diff
@@ -1 +1,320 @@
-export {}
+import { OpenSeaAccountURL, OPENSEA_API_KEY, OpenSea_API_URL } from './constants'
+import type { OpenSeaAssetContract, OpenSeaAssetEvent, OpenSeaResponse, OpenSeaCollection } from './types'
+import urlcat from 'urlcat'
+import { head, uniqBy } from 'lodash-unified'
+import type { AssetCollection, AssetOrder, NFTAsset, NFTHistory, OrderSide } from '../types'
+import { getOrderUnitPrice, getOrderUSDPrice, toTokenDetailed } from '../utils'
+import fromUnixTime from 'date-fns/fromUnixTime'
+
+import BigNumber from 'bignumber.js'
+import {
+    ChainId,
+    createERC721ContractDetailed,
+    createERC721Token,
+    ERC721TokenDetailed,
+    EthereumTokenType,
+    isSameAddress,
+} from '@masknet/web3-shared-evm'
+async function fetchAsset(url: string, chainId: ChainId) {
+    if (![ChainId.Mainnet, ChainId.Rinkeby].includes(chainId)) return
+
+    const response = await (
+        await fetch(url, {
+            method: 'GET',
+            mode: 'cors',
+            headers: {
+                'x-api-key': OPENSEA_API_KEY,
+            },
+        })
+    ).json()
+
+    return response
+}
+
+export async function getNFTsPaged(from: string, opts: { chainId?: ChainId; page?: number; size?: number }) {
+    const { chainId = ChainId.Mainnet, page = 0, size = 50 } = opts
+    const params = new URLSearchParams()
+    params.append('owner', from.toLowerCase())
+    params.append('limit', String(size))
+    params.append('offset', String(size * page))
+
+    const asset = await fetchAsset(`${OpenSea_API_URL}/api/v1/assets?${params.toString()}`, chainId)
+    if (!asset) return []
+
+    return asset.assets
+        .filter(
+            (x: OpenSeaResponse) =>
+                ['non-fungible', 'semi-fungible'].includes(x.asset_contract.type) ||
+                ['ERC721', 'ERC1155'].includes(x.asset_contract.schema_name),
+        )
+        .map((asset: OpenSeaResponse) => createERC721TokenAsset(from, asset.token_id, chainId, asset))
+}
+
+function createERC721ContractDetailedFromAssetContract(
+    address: string,
+    chainId: ChainId,
+    assetContract?: OpenSeaAssetContract,
+) {
+    return createERC721ContractDetailed(chainId, address, assetContract?.name ?? 'unknown name', '')
```

Comment:
Please use the same name and symbol fallback. FYI, here is the original creator method. 

```ts
export function createERC721ContractDetailed(
    chainId: ChainId,
    address: string,
    name = 'Unknown Token',
    symbol = 'UNKNOWN',
    baseURI?: string,
    iconURL?: string,
): ERC721ContractDetailed {
    return {
        type: EthereumTokenType.ERC721,
        chainId,
        address,
        name,
        symbol,
        baseURI,
        iconURL,
    }
}
```